src/kubernetes.cc

```diff
@@ -961,6 +961,18 @@ json::value KubernetesReader::FindTopLevelOwner(
   return FindTopLevelOwner(ns, GetOwner(ns, ref->As<json::Object>()));
 }
 
+bool KubernetesReader::ValidateConfiguration() const {
+  try {
+    json::value raw_node = QueryMaster(
+        std::string(kKubernetesEndpointPath) + "/nodes/" + CurrentNode());
```

Comment:
Should also validate `config_.KubernetesPodLabelSelector()`.

Reply:
This interaction is just validating that we can communicate with the master, adding a secondary request for pods is redundant, and would be more symptomatic of a failure inside of kubernetes, or a transient failure between requests.

Reply:
If so, then this is redundant, since `CurrentNode` issues two `QueryMaster` calls. So just calling `CurrentNode` is more than enough. Or you can call `QueryMaster` with a filter for one (random) node...

Reply:
I think we should continue to query for something like nodes, as the node name can be overridden by configuration, which would prevent the master interaction at all.

Reply:
Done.